File: app/models/qdm/guidance_response.rb
module QDM
  # app/models/qdm/guidance_response.rb
  class GuidanceResponse < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :requestIdentifier
    field :identifier
    field :module
    field :status
    field :subject
    field :encounter
    field :occurrenceDateTime
    field :performer
    field :reasonQDM::Code
    field :reasonReference
    field :note
    field :evaluationMessage
    field :outputParameters
    field :result
    field :dataRequirement
    field :qdmVersion, type: String, default: '4.0.0'
  end
end

File: app/assets/javascripts/GuidanceResponse.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const GuidanceResponseSchema = DataElementSchema({
  requestIdentifier: Any,
  identifier: Any,
  module: Any,
  status: Any,
  subject: Any,
  encounter: Any,
  occurrenceDateTime: Any,
  performer: Any,
  reasonCode: Any,
  reasonReference: Any,
  note: Any,
  evaluationMessage: Any,
  outputParameters: Any,
  result: Any,
  dataRequirement: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'GuidanceResponse' },

});

module.exports.GuidanceResponseSchema = GuidanceResponseSchema;
module.exports.GuidanceResponse = mongoose.model('GuidanceResponse', GuidanceResponseSchema);
